internal/output/output_result.go

```diff
@@ -0,0 +1,538 @@
+		if vuln.FixedVersion == VersionUnsupported {
+			return UnfixedDescription
+		}
+
+		// Skip vulnerabilities without a fixed version.
+		if vuln.FixedVersion == UnfixedDescription {
+			continue
+		}
+
+		if maxFixVersion == "" {
+			maxFixVersion = vuln.FixedVersion
+			vp = semantic.MustParse(maxFixVersion, ecosystemPrefix)
+
+			continue
+		}
+
+		// Update if the current vulnerability's fixed version is higher
+		if vp.CompareStr(vuln.FixedVersion) < 0 {
+			maxFixVersion = vuln.FixedVersion
+			vp = semantic.MustParse(maxFixVersion, ecosystemPrefix)
+		}
+	}
+
+	// Default to UnfixedDescription if no fix version is found.
+	if maxFixVersion == "" {
+		maxFixVersion = UnfixedDescription
+	}
+
+	return maxFixVersion
+}
+
+func updateVulnCount(original OutputVulnCount, newAdded OutputVulnCount) OutputVulnCount {
+	original.SeverityCount = addSeverityCount(original.SeverityCount, newAdded.SeverityCount)
+	original.CallAnalysisCount = addCallAnalysisCount(original.CallAnalysisCount, newAdded.CallAnalysisCount)
+	original.FixableCount = addFixableCount(original.FixableCount, newAdded.FixableCount)
+
+	return original
+}
+
+func addSeverityCount(original OutputSeverityCount, newAdded OutputSeverityCount) OutputSeverityCount {
+	original.Critical += newAdded.Critical
+	original.High += newAdded.High
+	original.Medium += newAdded.Medium
+	original.Low += newAdded.Low
+	original.Unknown += newAdded.Unknown
+
+	return original
+}
+
+func addCallAnalysisCount(original OutputCallAnalysisCount, newAdded OutputCallAnalysisCount) OutputCallAnalysisCount {
+	original.Called += newAdded.Called
+	original.Uncalled += newAdded.Uncalled
+
+	return original
+}
+
+func addFixableCount(original OutputFixableCount, newAdded OutputFixableCount) OutputFixableCount {
+	original.Fixed += newAdded.Fixed
+	original.UnFixed += newAdded.UnFixed
+
+	return original
+}
+
+func increaseSeverityCount(severityCount OutputSeverityCount, typeName string) OutputSeverityCount {
+	switch typeName {
+	case "CRITICAL":
+		severityCount.Critical += 1
+	case "HIGH":
+		severityCount.High += 1
+	case "MEDIUM":
+		severityCount.Medium += 1
+	case "LOW":
+		severityCount.Low += 1
+	case "UNKNOWN":
+		severityCount.Unknown += 1
+	}
+
+	return severityCount
+}
+
+func isOSEcosystem(ecosystem string) bool {
+	for _, image := range baseImages {
+		if strings.HasPrefix(ecosystem, image) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func getAllLayerInfo(result []OutputEcosystemResult) []OutputLayerInfo {
+	layerMap := make(map[string]string)
+	layerCount := make(map[string]OutputVulnCount)
+	layerIndex := 0
+
+	for _, ecosystem := range result {
+		for _, source := range ecosystem.Sources {
+			for _, packageInfo := range source.Packages {
+				layerID := packageInfo.LayerDetail.LayerID
+				layerCommand := packageInfo.LayerDetail.LayerCommand
+
+				// Check if this layer ID and command combination is already in the map
+				if _, ok := layerMap[layerID]; !ok {
```

Comment:
Hmm looking at this some more, does the if statement need to be here at all? Feel like we can just remove it since both sides of the statement basically does the same thing

Reply:
Yeah good catch